news-bundle/src/Controller/Page/NewsFeedController.php

```diff
@@ -66,16 +66,18 @@ public function __invoke(Request $request, PageModel $pageModel): Response
         $dispatcher = $this->container->get('event_dispatcher');
         $dispatcher->dispatch($event);
 
-        if (null !== ($articles = $event->getArticles())) {
-            foreach ($articles as $article) {
-                $event = new TransformArticleForFeedEvent($article, $feed, $pageModel, $request, $baseUrl);
-                $dispatcher->dispatch($event);
+        foreach ($event->getArticles() ?? [] as $article) {
+            $event = new TransformArticleForFeedEvent($article, $feed, $pageModel, $request, $baseUrl);
+            $dispatcher->dispatch($event);
 
-                $feed->add($event->getItem());
-
-                $this->tagResponse($article);
-                $this->tagResponse('contao.db.tl_news_archive.'.$article->pid);
+            if (!$item = $event->getItem()) {
+                continue;
             }
+
+            $feed->add($item);
```

Comment:
Should it not be like this?

```suggestion
            if ($item = $event->getItem()) {
                $feed->add($item);
            }
```

Otherwise 

```php
$this->tagResponse($article);
$this->tagResponse('contao.db.tl_news_archive.'.$article->pid);
```

are not executed if there are no items.

Reply:
But that's how it should be, no?

Reply:
Your change affects `$item` but the `tagResponse()` calls affect `$article`, so I'm not sure.

Reply:
Yes, we use `$article` because our response tagger automatically can tag the response for a `Contao\Model` object. It cannot do that with a Feed item object.

Reply:
Yes, and if we don't tag the response with the article ID – even if there are no items – how is the cache supposed to know which pages to invalidate when an item is added? @Toflar /cc

Reply:
The back end also invalidates `contao.db.tl_news_archive.X` in that case, so that's fine.

There is still one improvement that can be done here regarding the cache tags, but it has nothing to do with this PR or the issue that this PR fixes.

Reply:
» https://github.com/contao/contao/pull/8392

Reply:
Exactly, we've added invalidating the parent records for exactly that use case.